### PR TITLE
[reboot script] add -h, -? options

### PR DESCRIPTION
**Pull request by yxieca**
**- What I did**
Improve reboot script. Fixing issue https://github.com/Azure/sonic-buildimage/issues/2264

**- How I did it**
Added -h/-?/-x options

**- How to verify it**
Tested on DUT.

### CODE REVIEW

File: scripts/reboot [hunk 1]
```diff
@@ -30,6 +30,37 @@ function clear_warm_boot()
...
 }
 
+SCRIPT=$0
+
+function show_help_and_exit()
+{
+    echo "Usage ${SCRIPT} [options]"
+    echo "    Available options:"
+    echo "        -h -? : getting this help"
+    echo "        -x    : execute with -x flag"
```
Comment (qiluo-msft) [inline scripts/reboot]: Another option `bash -x reboot`. Do you really need this as a parameter?

Comment (yxieca) [reply scripts/reboot]: Not 100% sure. Maybe someone wants to watch reboot process? Feel a bit wired only adding help :-)

Comment (yxieca) [reply scripts/reboot]: Removed -x now.

File: scripts/reboot [hunk 2]
```diff
@@ -30,6 +30,36 @@ function clear_warm_boot()
     /sbin/kexec -u || /bin/true
 }
 
+SCRIPT=$0
+
+function show_help_and_exit()
+{
+    echo "Usage ${SCRIPT} [options]"
+    echo "    Request rebooting the device. Invoke platform tool when available."
```
Comment (jleveque) [inline scripts/reboot]: Suggest change "platform tool" to "platform-specific tool"

File: scripts/reboot [hunk 3]
```diff
@@ -30,6 +30,36 @@ function clear_warm_boot()
...
+}
+
+function parse_options()
+{
+    while getopts "h?" opt; do
+        case ${opt} in
+            h )
+                show_help_and_exit
+                ;;
+            \? )
```
Comment (jleveque) [inline scripts/reboot]: Suggest combining these case statements as follows:
```
    case ${opt} in
        h|\?)
        show_help_and_exit
        ;;
    esac
```

Please test and confirm that it works as expected.

File: scripts/reboot [hunk 4]
```diff
@@ -30,6 +30,36 @@ function clear_warm_boot()
...
+SCRIPT=$0
+
+function show_help_and_exit()
+{
+    echo "Usage ${SCRIPT} [options]"
+    echo "    Request rebooting the device. Invoke platform tool when available."
+    echo "    This script will shutdown syncd before rebooting."
+    echo " "
+    echo "    Available options:"
+    echo "        -h -? : getting this help"
```
Comment (jleveque) [inline scripts/reboot]: Suggest changing line to
```
    echo "        -h, -? : Output this help message"
```

File: scripts/reboot (hunk 2 above)
Comment (yxieca) [reply scripts/reboot]: Thanks! Done.

File: scripts/reboot (hunk 3 above)
Comment (yxieca) [reply scripts/reboot]: Thanks! Done.

File: scripts/reboot (hunk 4 above)
Comment (yxieca) [reply scripts/reboot]: Thanks! Done.

**Pull request merged**